### ISSUE TITLE
Python 3.x error when no sub-command

### ISSUE THREAD

**Issue by timcera** [rubik/mando#28]
Python 2.7

```
> tstoolbox
usage: tstoolbox [-h]
                 {fill,createts,filter,read,date_slice,describe,peak_detection,convert,equation,pick,stdtozrxp,tstopickle,accumulate,rolling_window,aggregate,clip,add_trend,remove_trend,calculate_fdc,stack,unstack,plot,dtw,pca,normalization}
                 ...
tstoolbox: error: too few arguments
```

Python 3.4:

```
> tstoolbox
Traceback (most recent call last):
  File "/home/tim/anaconda/envs/python3.4/bin/tstoolbox", line 9, in <module>
    load_entry_point('tstoolbox==0.9.9.6', 'console_scripts', 'tstoolbox')()
  File "/home/tim/anaconda/envs/python3.4/lib/python3.4/site-packages/tstoolbox/tstoolbox.py", line 2491, in main
    mando.main()
  File "/home/tim/anaconda/envs/python3.4/lib/python3.4/site-packages/mando/core.py", line 142, in __call__
    self.execute(sys.argv[1:])
  File "/home/tim/anaconda/envs/python3.4/lib/python3.4/site-packages/mando/core.py", line 136, in execute
    command, a = self.parse(args)
  File "/home/tim/anaconda/envs/python3.4/lib/python3.4/site-packages/mando/core.py", line 123, in parse
    command = arg_map.pop(_DISPATCH_TO)
KeyError: '_dispatch_to'
```

**Comment by rubik** [comment]
Uhm, this is odd. I'll look into it, thanks!

**Comment by rubik** [comment]
Ok, this is most definitely caused by a change in argparse. In the older version, it raised `SystemExit` when no arguments were passed. Now it doesn't and just returns an empty namespace.

**Comment by rubik** [comment]
Just as I thought. This is what happens in Python 2:
https://hg.python.org/cpython/file/2.7/Lib/argparse.py#l1949

But in Python 3 that error is never raised (there is no such check). That line (and just that line) was deleted:
https://github.com/python/cpython/blob/master/Lib/argparse.py#L1975

I don't think that this could be considered an argparse bug. The change seems intentional, since that error has been replaced with
https://github.com/python/cpython/blob/master/Lib/argparse.py#L1995

Now, a possible solution would be to automatically print the usage when no arguments are present, but then one loses the possibility to customize the parser. What do you suggest?

**Comment by timcera** [comment]
I don't have a feel of how many mando users have customized the parser.  I suspect zero or very few, but you may have a better idea if you have received e-mails about it.

There is a situation that might not be handled correctly.  What if all arguments are optional?  You wouldn't want the usage displayed.  That just occurred to me, but I don't have a handy test.

**Comment by MarioSchwalbe** [comment]
There may be an easy fix. Inserting
`
if _DISPATCH_TO not in arg_map:
    self.parser.error("too few arguments")
`
right after calling `self.parser.parse_args(args)` can emulate the old behavior.

**Comment by rubik** [comment]
Fixed by Mario in #32.
